enumerables.rb

```diff
@@ -0,0 +1,135 @@
+module Enumerable
+
+  def my_each
+    return p to_enum(__method__) unless block_given?
+      i = 0
+      while i < self.size
+        yield(self[i])
+        i += 1
+      end
+      return to_enum
+  end
+
+  def my_each_with_index
+      if block_given?
+        ind = 0
+        size.times do
+          yield(self[ind], ind)
+          ind += 1
+        end
+      else
+        return to_enum(__method__)
+      end
+    end
+
+  def my_select
+    return p to_enum(__method__) unless block_given?
+      array = []
+      i = 0
+      while i < self.size
+          array << i if yield(self[i])
+          i +=1                
+      end
+      return to_enum
+  end
```

Comment:
* [ ] my_select returns an array containing all elements of enum for which the given block returns a true value
  ```
    block = proc {|x| x>4}
    array = [1,2,5,8,7,98,4]
    p array.my_select(&block) == array.select(&block) # false
  ```

- There is two issues with the code:
1- the array should take the elements that satisfy a condition not their indexes. 
2- You shouldn't worry about returning an Enumerator at the end, you just need to return the "array" variable.